frodo.js

```diff
@@ -3,70 +3,71 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.